File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Functionality/MaximumPropertyPage.vb
Namespace CompanionKit.Controls.ProgressBar.Functionality

	Public Class MaximumPropertyPage

        Private mintNumber As Integer = 0

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        ''' <summary>
        ''' Handle Tick event of the timer that increase value of the ProgressBars.
        ''' </summary>
        Private Sub mobjIncrementTimer_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjIncrementTimer.Tick
            ' Compute new value of the the demonstrated ProgressBars
            Me.mintNumber = (Me.mintNumber + 5)
            If (Me.mintNumber > 100) Then
                Me.mintNumber = 0
            End If

            ' Set value for the demonstrated ProgressBars 
            Me.mobjDemoMax100ProgressBar.Value = Me.mintNumber

            Me.mobjDemoMax25ProgressBar.Value = (Me.mintNumber Mod Me.mobjDemoMax25ProgressBar.Maximum)

        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Functionality/MaximumPropertyPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.ProgressBar.Functionality
{
    public partial class MaximumPropertyPage : UserControl
    {
        private int mintNumber = 0;

        public MaximumPropertyPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handle Tick event of the timer that increase value of the ProgressBars.
        /// </summary>
        private void mobjIncrementTimer_Tick(object sender, EventArgs e)
        {
            // Compute new value of the the demonstrated ProgressBars
            mintNumber += 5;
            if (mintNumber > 100)
            {
                mintNumber = 0;
            }

            // Set value for the demonstrated ProgressBars
            this.mobjDemoMax100ProgressBar.Value = mintNumber;
            this.mobjDemoMax25ProgressBar.Value = mintNumber % mobjDemoMax25ProgressBar.Maximum;
        }
    }
}